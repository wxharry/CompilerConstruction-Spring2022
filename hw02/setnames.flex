

%%

%public
%class Subst
%standalone

%unicode

%{
  String[] names;
%}

LineTerminator = \r|\n|\r\n

%%
beginnames{LineTerminator}([A-Za-z]+{LineTerminator})*endnames{LineTerminator} {
  String[] lines = yytext().split("\r\n");
  for(int i=0; i<lines.length; i++){
    System.out.println();
  } 
  names = new String[lines.length-2];
  for(int i=0; i<names.length; i++){
    names[i] = lines[i+1];
  }
}
"names"     {System.out.print(String.join(" and ", names));}
