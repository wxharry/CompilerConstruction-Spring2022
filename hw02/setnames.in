beginnames
Alice
Bob
Chara
endnames
Yesterday I met names. names live nearby.