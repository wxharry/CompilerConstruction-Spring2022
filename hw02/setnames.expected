




Yesterday I met Alice and Bob and Chara. Alice and Bob and Chara live nearby.